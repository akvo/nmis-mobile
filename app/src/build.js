import { SERVER_URL } from '@env';
import build_json from './build.json';

const defaultBuildParams = {
  authenticationType: ['code_assignment', 'username', 'password'],
  ...build_json,
  serverURL: SERVER_URL,
};

export default defaultBuildParams;
