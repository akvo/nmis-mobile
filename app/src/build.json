{
  "authenticationType": ["code_assignment", "username", "password"],
  "serverURL": "http://192.168.21.252:8080",
  "debugMode": false,
  "dataSyncInterval": 300,
  "errorHandling": true,
  "loggingLevel": "verbose",
  "appVersion": "1.2.0",
  "lang": "en"
}
