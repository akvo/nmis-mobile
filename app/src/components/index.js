export { default as Stack } from './Stack';
export { default as Carousel } from './Carousel';
export { default as DividerLayout } from './DividerLayout';
