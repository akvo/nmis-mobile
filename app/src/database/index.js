export * from './conn';
export * from './query';
