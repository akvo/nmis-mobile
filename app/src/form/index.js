export { default as Webform } from './Webform';
