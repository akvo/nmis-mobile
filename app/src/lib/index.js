export { default as api } from './api';
export { default as loc } from './loc';
export { default as notification } from './notification';
export { default as backgroundTask } from './background-task';
export { default as cascades } from './cascades';
