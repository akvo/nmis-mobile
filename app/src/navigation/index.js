import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { HomePage, FormActionPage, FormDataPage, GetStartedPage, AuthFormPage } from '../pages';

const Stack = createNativeStackNavigator();

const RootNavigator = () => {
  return (
    <Stack.Navigator screenOptions={{ headerShown: false }}>
      <Stack.Screen name="GetStarted" component={GetStartedPage} />
      <Stack.Screen name="AuthForm" component={AuthFormPage} />
      <Stack.Screen name="Home" component={HomePage} />
      <Stack.Screen name="FormAction" component={FormActionPage} />
      <Stack.Screen name="FormData" component={FormDataPage} />
    </Stack.Navigator>
  );
};

const Navigation = (props) => {
  return (
    <NavigationContainer {...props}>
      <RootNavigator />
    </NavigationContainer>
  );
};

export default Navigation;
