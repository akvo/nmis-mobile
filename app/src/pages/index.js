export { default as HomePage } from './Home';
export { default as FormActionPage } from './FormAction';
export { default as FormDataPage } from './FormData';
export { default as GetStartedPage } from './GetStarted';
export { default as AuthFormPage } from './AuthForm';
export { default as SettingsPage } from './Settings';
export { default as SettingsFormPage } from './Settings/Form';
