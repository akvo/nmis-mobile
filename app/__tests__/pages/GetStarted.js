import React from 'react';
import renderer from 'react-test-renderer';
import { GetStartedPage } from '../../src/pages';

describe('GetStartedPage', () => {
  test('renders correctly', () => {
    const tree = renderer.create(<GetStartedPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
