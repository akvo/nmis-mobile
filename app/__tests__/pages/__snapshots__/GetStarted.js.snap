// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetStartedPage renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f9fafb",
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
      "gap": 36,
      "justifyContent": "center",
      "paddingHorizontal": 16,
    }
  }
  testID="center-layout"
>
  <View
    style={
      {
        "backgroundColor": "#e5e7eb",
        "borderRadius": 4,
        "height": 110,
        "width": 110,
      }
    }
    testID="image-skeleton"
  />
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
      }
    }
    testID="center-layout-items"
  >
    <Text
      accessibilityRole="text"
      style={
        {
          "color": "#242424",
          "fontSize": 28.285714285714285,
        }
      }
    >
      Splash Screen
    </Text>
  </View>
  <Text
    accessibilityRole="text"
    style={
      {
        "color": "#242424",
      }
    }
  >
    Welcome
  </Text>
</View>
`;
